metrics/time/launch_times.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#  Description of the test:
+#  This test takes a number of time measurements through the complete
+#  launch/shutdown cycle of a single container.
+#  From those measurements it derives a number of time measures, such as:
+#   - time to payload execution
+#   - time to get to VM kernel
+#   - time in VM kernel boot
+#   - time to quit
+#   - total time (from launch to finished)
+#
+# Note, the <image> used for this test must support the full 'date' command
+# syntax - the date from busybox for isntance *does not* support this, so
+# will not work with this test.
+#
+# Note, this test launches a single container at a time, that quits - thus,
+# this test measures times for the 'first container' only. This test does
+# not look for any scalability slowdowns as the number of running containers
+# increases for instance - that is handled in other tests
+
+set -e
+
+if [ $# -ne 3 ] && [ $# -ne 4 ]; then
+	echo >&2 "Usage: $0 <image> <runtime> <times to run> (<scaling>)"
+	exit 1
+fi
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+IMAGE="$1"
+RUNTIME="$2"
+TIMES="$3"
+SCALING="$4"
+
+# Calculating the kernel time from dmesg stamps only really works for VM
+# based runtimes - we dynamically enable it if we find we are using a known
+# VM runtime
+CALCULATE_KERNEL=
+
+# We set the generic name here, but we save the different time results separately,
+# and append the actual detail to the name at the time of saving...
+TEST_NAME="boot times"
+# If we are scaling, note that in the name
+if [ ! -z "$SCALING" ]; then
```

Comment:
I think you can use ``` [ -n  "$SCALING" ] ```

Reply:
You could event drop the `if` for brevity:

```
[ -n "$SCALING" ] && TEST_NAME="${TEST_NAME} scaling"
```

Reply:
Indeed, I probably can use `-n`. I'll go have a trawl and try. I looked at that briefly last week, but took the easy known to work route ;-)